core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2420,6 +2420,44 @@ private[spark] object Utils extends Logging {
   }
 }
 
+private[spark] class CallerContext(
+   appName: Option[String] = None,
+   appID: Option[String] = None,
+   appAttemptID: Option[String] = None,
+   jobID: Option[Int] = None,
+   stageID: Option[Int] = None,
+   stageAttemptId: Option[Int] = None,
+   taskId: Option[Long] = None,
+   taskAttemptNumber: Option[Int] = None) extends Logging {
+
+   val AppName = if (appName.isDefined) s"_AppName_${appName.get}" else ""
```

Comment:
I'm not sure what you mean here there is no information about the application.  its pretty clear that the application id in your example is: application_1473908768790_0007.  We don't need the extra 6 characters AppID_ to tell us that